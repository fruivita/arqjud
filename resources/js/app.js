/**
 * @see https://laravel.com/docs/9.x/vite
 */

import Layout from '@/Shared/Layout.vue';
import { createInertiaApp, Link } from '@inertiajs/inertia-vue3';
import { InertiaProgress } from '@inertiajs/progress';
import { resolvePageComponent } from 'laravel-vite-plugin/inertia-helpers';
import { createPinia } from 'pinia';
import { createApp, h } from 'vue';
import Toast from 'vue-toastification';
import 'vue-toastification/dist/index.css';
import '../css/app.css';

createInertiaApp({
    resolve: async (name) => {
        let page = (
            await resolvePageComponent(`./Pages/${name}.vue`, import.meta.glob('./Pages/**/*.vue'))
        ).default;

        if (page.layout === undefined) {
            page.layout = Layout;
        }

        return page;
    },
    setup({ el, App, props, plugin }) {
        createApp({ render: () => h(App, props) })
            .use(plugin)
            .use(createPinia())
            .use(Toast)
            .mixin({ methods: { route } })
            .component('InertiaLink', Link)
            .mount(el);
    },
});

InertiaProgress.init({
    // The delay after which the progress bar will
    // appear during navigation, in milliseconds.
    delay: 250,

    // The color of the progress bar.
    color: 'red',

    // Whether to include the default NProgress styles.
    includeCSS: true,

    // Whether the NProgress spinner will be shown.
    showSpinner: true,
});
