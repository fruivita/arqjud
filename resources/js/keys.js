/**
 * Chaves para provide/inject e outras constantes globais
 *
 * @see https://vuejs.org/guide/components/provide-inject.html#working-with-symbol-keys
 */

export const perPageKey = Symbol();
export const updatePerPageKey = Symbol();
export const exibirModalKey = Symbol();
export const fecharModalKey = Symbol();
export const mascaraCNJ = '#######-##.####.#.##.####';
