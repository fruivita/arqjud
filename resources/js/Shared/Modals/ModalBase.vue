<!--
    Modal padrão que deve ser utilizado pelos modais específicos.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
    @link https://www.thisdot.co/blog/provide-inject-api-with-vue-3
 -->

<script setup>
import { exibirModalKey, fecharModalKey } from '@/keys';
import ButtonText from '@/Shared/Buttons/ButtonText.vue';
import { useTranslationsStore } from '@/Stores/TranslationsStore';
import { onClickOutside } from '@vueuse/core';
import { inject, ref } from 'vue';

const __ = useTranslationsStore().__;

const exibirModal = inject(exibirModalKey);
const fecharModal = inject(fecharModalKey);

const container = ref(null);

onClickOutside(container, fecharModal);
</script>

<template>
    <Transition
        enter-from-class="opacity-0 scale-150"
        enter-to-class="opacity-100 scale-100"
        enter-active-class="transition duration-300 transform-gpu"
        leave-active-class="transition duration-200 transform-gpu"
        leave-from-class="opacity-100 scale-100"
        leave-to-class="opacity-0 scale-150"
    >
        <div
            v-if="exibirModal"
            class="fixed inset-0 z-20 grid place-items-center bg-primaria-100/80 text-xl text-primaria-900 dark:bg-secundaria-900/80 dark:text-secundaria-50"
        >
            <article class="w-full divide-y lg:w-8/12" ref="container">
                <header class="rounded-t-lg bg-primaria-300 p-3 dark:bg-secundaria-700">
                    <h6 class="text-2xl font-bold">
                        <slot name="header"></slot>
                    </h6>
                </header>

                <div
                    class="max-h-48 overflow-y-auto bg-primaria-50 px-3 py-6 dark:bg-secundaria-900 lg:px-24"
                >
                    <slot name="body"></slot>
                </div>

                <footer
                    class="flex flex-col justify-end space-x-0 space-y-3 rounded-b-lg bg-primaria-300 p-3 dark:bg-secundaria-700 lg:flex-row lg:items-center lg:space-x-3 lg:space-y-0"
                >
                    <slot name="footer"></slot>

                    <ButtonText
                        :texto="__('Cancelar')"
                        @click="fecharModal"
                        dusk="btn-cancelar"
                        especie="inacao"
                        icone="x-circle"
                        type="button"
                    />
                </footer>
            </article>
        </div>
    </Transition>
</template>
