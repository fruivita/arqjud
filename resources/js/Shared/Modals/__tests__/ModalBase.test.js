/**
 * Testes para o componente ModalBase.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import { exibirModalKey, fecharModalKey } from '@/keys.js';
import ModalBase from '@/Shared/Modals/ModalBase.vue';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test, vi } from 'vitest';

let mountFunction;
let pinia;

beforeEach(() => {
    pinia = (options = {}) =>
        createTestingPinia({
            stubActions: false,
            initialState: {
                StatusRequisicaoStore: { ...options },
            },
        });

    mountFunction = (options = {}) => {
        return mount(ModalBase, { ...options });
    };
});

describe('ModalBase', () => {
    const opcoes = [true, false];

    test.each(opcoes)(
        'renderiza e exibe o componente de acordo com o valor provido pelo componente pai',
        (opcao) => {
            expect(
                mountFunction({
                    global: {
                        provide: {
                            [exibirModalKey]: opcao,
                            [fecharModalKey]: vi.fn(),
                        },
                        plugins: [pinia({ processando: false })],
                    },
                    slots: {
                        header: 'foo',
                        body: 'bar',
                        footer: 'baz',
                    },
                }).html()
            ).toMatchSnapshot();
        }
    );
});
