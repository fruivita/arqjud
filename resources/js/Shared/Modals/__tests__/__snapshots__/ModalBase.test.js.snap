// Vitest Snapshot v1

exports[`ModalBase > renderiza e exibe o componente de acordo com o valor provido pelo componente pai 1`] = `
"<transition-stub enterfromclass=\\"opacity-0 scale-150\\" entertoclass=\\"opacity-100 scale-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100 scale-100\\" leavetoclass=\\"opacity-0 scale-150\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <div class=\\"fixed inset-0 z-20 grid place-items-center bg-primaria-100/80 text-xl text-primaria-900 dark:bg-secundaria-900/80 dark:text-secundaria-50\\">
    <article class=\\"w-full divide-y lg:w-8/12\\">
      <header class=\\"rounded-t-lg bg-primaria-300 p-3 dark:bg-secundaria-700\\">
        <h6 class=\\"text-2xl font-bold\\">foo</h6>
      </header>
      <div class=\\"max-h-48 overflow-y-auto bg-primaria-50 px-3 py-6 dark:bg-secundaria-900 lg:px-24\\">bar</div>
      <footer class=\\"flex flex-col justify-end space-x-0 space-y-3 rounded-b-lg bg-primaria-300 p-3 dark:bg-secundaria-700 lg:flex-row lg:items-center lg:space-x-3 lg:space-y-0\\">baz<button class=\\"bg-gray-500 text-gray-50 ring-gray-600 hover:bg-gray-600 focus:bg-gray-600 inline-flex transform-gpu items-center justify-center space-x-4 rounded-lg px-4 py-2 outline-none transition duration-300 hover:ring-4 focus:ring-4 disabled:cursor-not-allowed disabled:opacity-50\\" type=\\"button\\" dusk=\\"btn-cancelar\\"><span dusk=\\"texto\\">Cancelar</span><svg class=\\"w-6 h-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
            <path d=\\"M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z\\"></path>
            <path d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path>
          </svg></button></footer>
    </article>
  </div>
</transition-stub>"
`;

exports[`ModalBase > renderiza e exibe o componente de acordo com o valor provido pelo componente pai 2`] = `
"<transition-stub enterfromclass=\\"opacity-0 scale-150\\" entertoclass=\\"opacity-100 scale-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100 scale-100\\" leavetoclass=\\"opacity-0 scale-150\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <!--v-if-->
</transition-stub>"
`;
